src/hotspot/share/opto/loopTransform.cpp

```diff
@@ -124,7 +124,8 @@ void IdealLoopTree::compute_trip_count(PhaseIdealLoop* phase) {
     jlong limit_con = (stride_con > 0) ? limit_type->_hi : limit_type->_lo;
     int stride_m = stride_con - (stride_con > 0 ? 1 : -1);
     jlong trip_count = (limit_con - init_con + stride_m)/stride_con;
-    if (trip_count > 0 && (julong)trip_count < (julong)max_juint) {
+    trip_count = MAX2(trip_count, (jlong)1);
```

Comment:
Add comment here explaining this case (one trip when init >= limit).

BTW, this optimization seems only works for INT iv loops and not LONG. Do you plan to implement for LONG?

Reply:
> Add comment here explaining this case (one trip when init >= limit).

Ok. Do you still think we need extra tests?

> BTW, this optimization seems only works for INT iv loops and not LONG. Do you plan to implement for LONG?

I thought about it but it's not straightforward because the current code uses long to avoid overflow. I would also like to rework the parallel iv code so it works with LONG loops. I don't have time right now, thought.

Note also, that because of the iv phi Value(), I would expect some of the single iteration loops to be optimized even without a working IdealLoopTree::compute_trip_count(). That would be true for in and long loops.

Reply:
> > Add comment here explaining this case (one trip when init >= limit).
> 
> Ok. Do you still think we need extra tests?

Yes, would be nice to have tests for 0,1,2 iterations and all 3 types of loops: `for() {}`, `while() {}`, `do {} while()` to verify that they are converted to straight code. It could be done as separate RFE.

> 
> > BTW, this optimization seems only works for INT iv loops and not LONG. Do you plan to implement for LONG?
> 
> I thought about it but it's not straightforward because the current code uses long to avoid overflow. I would also like to rework the parallel iv code so it works with LONG loops. I don't have time right now, thought.
> 
> Note also, that because of the iv phi Value(), I would expect some of the single iteration loops to be optimized even without a working IdealLoopTree::compute_trip_count(). That would be true for in and long loops.

Got it. Thank you.

Reply:
> Yes, would be nice to have tests for 0,1,2 iterations and all 3 types of loops: `for() {}`, `while() {}`, `do {} while()` to verify that they are converted to straight code. It could be done as separate RFE.

I filed https://bugs.openjdk.java.net/browse/JDK-8262721
Sounds like a good candidate for the new testing framework that allows pattern matching on the IR.